src/mc_tasks/ImpedanceTask.cpp

```diff
@@ -0,0 +1,311 @@
+  logger.addLogEntry(name_ + "_deltaComplianceAccel", [this]() -> const sva::MotionVecd & { return deltaCompAccelW_; });
+
+  // desired values
+  logger.addLogEntry(name_ + "_desiredPose", [this]() -> const sva::PTransformd & { return desiredPoseW_; });
+  logger.addLogEntry(name_ + "_desiredVel", [this]() -> const sva::MotionVecd & { return desiredVelW_; });
+  logger.addLogEntry(name_ + "_desiredAccel", [this]() -> const sva::MotionVecd & { return desiredAccelW_; });
+
+  // wrench
+  logger.addLogEntry(name_ + "_targetWrench", [this]() -> const sva::ForceVecd & { return targetWrench_; });
+  logger.addLogEntry(name_ + "_measuredWrench", [this]() -> const sva::ForceVecd & { return measuredWrench_; });
+  logger.addLogEntry(name_ + "_filteredMeasuredWrench",
+                     [this]() -> const sva::ForceVecd & { return filteredMeasuredWrench_; });
+  logger.addLogEntry(name_ + "_cutoffPeriod", [this]() { return cutoffPeriod(); });
+}
+
+void ImpedanceTask::removeFromLogger(mc_rtc::Logger & logger)
+{
+  SurfaceTransformTask::removeFromLogger(logger);
+
+  // impedance parameters
+  logger.removeLogEntry(name_ + "_impedanceM");
+  logger.removeLogEntry(name_ + "_impedanceD");
+  logger.removeLogEntry(name_ + "_impedanceK");
+  logger.removeLogEntry(name_ + "_wrenchGain");
+
+  // compliance values
+  logger.removeLogEntry(name_ + "_deltaCompliancePose");
+  logger.removeLogEntry(name_ + "_deltaComplianceVel");
+  logger.removeLogEntry(name_ + "_deltaComplianceAccel");
+
+  // desired values
+  logger.removeLogEntry(name_ + "_desiredPose");
+  logger.removeLogEntry(name_ + "_desiredVel");
+  logger.removeLogEntry(name_ + "_desiredAccel");
+
+  // wrench
+  logger.removeLogEntry(name_ + "_targetWrench");
+  logger.removeLogEntry(name_ + "_measuredWrench");
+  logger.removeLogEntry(name_ + "_filteredMeasuredWrench");
+  logger.removeLogEntry(name_ + "_cutoffPeriod");
+}
+
+void ImpedanceTask::addToGUI(mc_rtc::gui::StateBuilder & gui)
+{
+  // Don't add SurfaceTransformTask because the target of SurfaceTransformTask should not be set by user
+  TrajectoryTaskGeneric<tasks::qp::SurfaceTransformTask>::addToGUI(gui);
+
+  gui.addElement({"Tasks", name_},
+                 // pose
+                 mc_rtc::gui::Transform("desiredPose",
+                                        [this]() -> const sva::PTransformd & { return this->desiredPose(); },
+                                        [this](const sva::PTransformd & pos) { this->desiredPose(pos); }),
+                 mc_rtc::gui::Transform("compliancePose", [this]() { return this->compliancePose(); }),
+                 mc_rtc::gui::Transform("pose", [this]() { return this->surfacePose(); }),
+                 // impedance parameters
+                 mc_rtc::gui::ArrayInput("impedanceM", {"cx", "cy", "cz", "fx", "fy", "fz"},
+                                         [this]() { return this->impedanceM().vector(); },
```

Comment:
FYI, you can return a `const sva::ForceVecd &` here as the GUI serializer knows how to transform it into an array

Reply:
And with #112 it's actually a little better to do so as this incurs no additional copy to get the vector

Reply:
> FYI, you can return a const sva::ForceVecd & here as the GUI serializer knows how to transform it into an array

Although `ForceVecd::vector()` does not return const reference, can we return const reference in GUI?
https://github.com/jrl-umi3218/SpaceVecAlg/blob/1668efd3f2647c6a850d268b73ce21930ebbeb38/src/SpaceVecAlg/ForceVec.h#L88